gen.py

```diff
@@ -97,93 +101,119 @@ def convert_svg2png(infile, outfile, w, h):
+            root = icon["linux"]["root"] + ".svg"
             output = "{0}/{1}/apps/".format(linux_dir, size)
-            if path.exists(source):
-                if icon["linux"].get("bfb") and (size == "48"):
-                    output_bfb = "{0}.png".format(icon["linux"].get("bfb"))
-                    convert_svg2png(source, output + output_bfb, 144, 144)
-                copy2(source, output + root)
-                for link in icon["linux"].get("symlinks", []):
-                    output_symlink = "{0}{1}.svg".format(output, link)
-                    try:
-                        symlink(root, output_symlink)
-                    except FileExistsError:
-                        continue
+            copy2(source, output + root)
+
+            # Symlinks to root icon
+            for link in icon["linux"].get("symlinks", []):
+                output_symlink = output + link + ".svg"
+                try:
+                    symlink(root, output_symlink)
+                except FileExistsError:
+                    continue
+
+            # Unity 7 BFB icons
+            if icon["linux"].get("bfb") and (size == "48"):
+                output_bfb = icon["linux"].get("bfb") + ".png"
+                convert_svg2png(source, output + output_bfb, 144, 144)
+
+
+# OSX Generation Code
 elif platform == "osx":
     print("\nGenerating OSX theme...")
-    ink_flag = call(['which', 'png2icns'], stdout=PIPE, stderr=PIPE)
-    if ink_flag != 0:
+
+    # OSX generation depends on libicns function png2icns
+    no_icns = call(["which", "png2icns"], stdout=PIPE, stderr=PIPE)
+    if no_icns:
         exit("You will need png2icns in order to generate OSX theme")
+
+    # Define and create theme locations
     osx_dir = "numix-{0}.icns".format(theme)
-    osx_sub_dirs = ["icns", "pngs", "vectors"]
-    for sub_dir in osx_sub_dirs:
+    for sub_dir in ["icns", "pngs", "vectors"]:
         mkdir("{0}/{1}".format(osx_dir, sub_dir))
+
     for icon_name, icon in icons.items():
         for output_icon in icon.get("osx", []):
+            # Check icon source exists before it's used
             source = "icons/{0}/48/{1}.svg".format(theme, icon_name)
-            output_svg = "{0}/vectors/{1}.svg".format(osx_dir, output_icon)
-            output_png = "{0}/pngs/{1}.png".format(osx_dir, output_icon)
-            output_icn = "{0}/icns/{1}.icn".format(osx_dir, output_icon)
-            if path.exists(source):
-                copy2(source, output_svg)
-                convert_svg2png(source, output_png, 1024, 1024)
-                call(["png2icns", output_icn, output_png],
-                     stdout=PIPE, stderr=PIPE)
+            if not path.exists(source):
+                continue
+
+            # Icon output locations
+            svg_icon = "{0}/vectors/{1}.svg".format(osx_dir, output_icon)
+            png_icon = "{0}/pngs/{1}.png".format(osx_dir, output_icon)
+            icn_icon = "{0}/icns/{1}.icn".format(osx_dir, output_icon)
```

Comment:
Not gonna lie, I renamed these variables *entirely* so that the final `call` didn't have to be split over two lines.